csharp/src/Apache.Arrow/Ipc/ArrowFileWriter.cs

```diff
@@ -101,7 +101,7 @@ private protected override void FinishedWritingRecordBatch(long bodyLength, long
             _currentRecordBatchOffset = -1;
         }
 
-        public async Task WriteFooterAsync(CancellationToken cancellationToken = default)
+        public async ValueTask WriteFooterAsync(CancellationToken cancellationToken = default)
```

Comment:
Consider leaving these non-generic ones as Task.

Reply:
For my learning, what's the reason? Is it because it will just use `Task.CompletedTask` under the covers and thus not need to allocate?

Reply:
https://devblogs.microsoft.com/dotnet/understanding-the-whys-whats-and-whens-of-valuetask/

In this particular case with this implementation, there's zero upside and only potential downside to using ValueTask:
- For synchronous completions with Task, the implementation would just return a cached successfully completed task, e.g. Task.CompletedTask.
- ValueTask is (in microbenchmarks) more expensive to await, because it's a struct wrapping a Task, so every operation requires a check to see what state it's in, and today at least the JIT isn't great about structs wrapping other things, in particular structs wrapping other structs, which is the situation you end up in with the struct awaiter wrapping the struct ValueTask.
- For a generic `ValueTask<T>`, it's generally worth it, because you avoid the allocation for a `Task<T>` assuming you couldn't cache one for every possible `T`.  But for `ValueTask`, for synchronous completions, there's only ever one possible thing.
- For asynchronous completions, today, async methods just fall back to using a `Task` and wrapping the `ValueTask` around it, so it's only potential overhead on top of using `Task` directly.

The only benefit to using `ValueTask` would be if you believe in the future you might choose to replace this implementation with one that used a pooled `IValueTaskSource` as the backing implementation for asynchronously completing operations and if such asynchronously completing operations were commonplace, on a hot path, and mattered for perf.